library/alloc/tests/task.rs

```diff
@@ -0,0 +1,34 @@
+use alloc::rc::Rc;
+use alloc::sync::Arc;
+use alloc::task::{LocalWake, Wake};
+use core::task::{LocalWaker, Waker};
+
+#[test]
+fn test_waker_will_wake_clone() {
+    struct NoopWaker;
+
+    impl Wake for NoopWaker {
+        fn wake(self: Arc<Self>) {}
+    }
+
+    let waker = Waker::from(Arc::new(NoopWaker));
+    let clone = waker.clone();
+
+    assert!(waker.will_wake(&clone));
+    assert!(clone.will_wake(&waker));
```

Comment:
These tests are testing things which are not guaranteed by the language, they just happen to be what rustc currently does. In particular, these tests [fail in Miri](https://github.com/rust-lang/miri-test-libstd/actions/runs/8135079336/job/22229021537).

Not sure if that is intentional; there's no comment here indicating such an intent.